tools/config/configure.R

```diff
@@ -279,8 +279,17 @@ if (is.null(find_nvcc(stop_if_missing = FALSE)) || !has_libcuml()) {
   wd <- getwd()
   on.exit(setwd(wd))
   setwd(pkg_root())
-  define(PKG_CPPFLAGS = normalizePath(file.path(getwd(), "src", "stubs")))
+  stubs_header_dir <- normalizePath(file.path(getwd(), "src", "stubs"))
+  writeLines(
+    c(
+      sprintf("PKG_CPPFLAGS = -I'%s'", stubs_header_dir),
+      # optimize for size
+      "strippedLib: $(SHLIB)",
+      "\tif test -e '/usr/bin/strip'; then /usr/bin/strip --strip-debug $(SHLIB); fi",
+      ".phony: strippedLib"
```

Comment:
Nothing interesting in this PR, aside from 2 things:

- If we are not linking to `libcuml++`, then some code paths still need to return model objects / `NumericMatrix`s / `NumericVector`s with correct attributes / dimensions to make CRAN happy
- Seems like even when building a `cuda.ml.so` shared lib not linked to `libcuml++`, the `libs` size will still reach 5+ MB, at which point a NOTE gets emitted, and it's probably best to avoid this type of NOTE, right? (from the standpoint of making CRAN happy) -- Anyhow, the only thing I can think of here is to throw away debugging symbols.

Reply:
In general I haven't had problems with this note in packages that include compiled code. For example, torch raises it since the first submission.

Reply:
Human feedback from CRAN: I'm actually not allowed to strip DSOs (still not sure which part of the CRAN policies mentioned this, but anyhow, not allowed 😅)